src/core/utils/net.ts

```diff
@@ -2,24 +2,51 @@ import chalk from "chalk";
 import net from "net";
 import ora from "ora";
 import waitOn from "wait-on";
+import getPort from "get-port";
+import { DEFAULT_CONFIG } from "../../config";
+import { confirmChooseRandomPort } from "../prompts";
 import { logger } from "./logger";
 
+const VALID_PORT_MIN = 1024;
+const VALID_PORT_MAX = 65535;
+
 /**
  * Check if a given remote address and port are accepting TCP connections.
  * @param Object host and port of the server to check.
- * @returns True if the given server is accepting TCP connections. False otherwise.
+ * @returns The resolved port number if the given server is accepting TCP connections. 0 if the port is already taken.
  */
-export function isAcceptingTcpConnections({ host = "127.0.0.1", port = 80 }: { host?: string; port?: number }) {
-  return new Promise<boolean>((resolve) => {
-    const socket = net.createConnection(port, host);
+export function isAcceptingTcpConnections({
+  host = DEFAULT_CONFIG.host,
```

Comment:
host/port should come from the commander default, so there should be no need to have a fallback wired to DEFAULT_CONFIG here again?